packages/react/document-context/src/document-context.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+
+// Use null as initial value to handle SSR safely
+const DocumentContext = React.createContext<Document | null>(null);
+
+interface DocumentProviderProps {
+  document?: Document;
+  children: React.ReactNode;
+}
+
+export function DocumentProvider({ document: doc, children }: DocumentProviderProps) {
+  const value = React.useMemo(
+    () => doc ?? (typeof document !== 'undefined' ? globalThis?.document : null),
```

Comment:
I don't think we need to fallback to the global document in this case. This is already handled in `useDocument`.

In either event, memoization in this context isn't really needed since the reference is outside the React tree and stable.